doc/node-devbox-setup.md

```diff
@@ -25,29 +25,63 @@ The SDK is entirely open-source (as you probably figured out already if you're r
 $ git clone https://github.com/azure/azure-iot-sdk-node
 ```
 
-The SDK ships in a few separate NPM packages that depend on each other. Once the repository has been cloned, some features span accross multiple packages so in order to build and test these features, we need to "link" those package together in a single environment. The following steps explain how to do that.
+The SDK ships in a few separate NPM packages that depend on each other. Once the repository has been cloned, some features span accross multiple packages so in order to build and test these features, we need to "link" those package together in a single environment. We use the awesome [lerna.js](https://lernajs.io) to do that, so you have to install this first.
+
+```
+npm install -g lerna
+```
+
+Once lerna is installed, you can set up your development environment by running the `bootstrap command` at the root of the repository: This will install all dependencies and link packages together.
+
+```
+lerna bootstrap
+```
+
+If you want to build/run the code, you'll need to compile the packages:
+
+```
+lerna run build
+```
+
+If you want to run the tests:
+
+```
+lerna run ci
+```
+
+and the end-to-end tests:
+
+```
+lerna run e2e
+```
+
+Please note that running tests and end-to-end tests require having an Azure IoT hub and an Azure IoT Hub Provisioning service setup and the proper environment variables configured. Here is the list of all the environment variables as well as how we use them:
+
+- **OPENSSL_CONF**: The SDK build script relies on OpenSSL to create certificates and keys so OpenSSL must be in the path and `OPENSSL_CONF` must be set to the path of your `openssl.cnf` configuration file
+- **IOTHUB_CONNECTION_STRING** must be set to a configuration string of your IoT Hub that has rights to create/delete devices and send messages to devices (typically, the one associated with the `iothubowner` policy or equivalent). Connection strings can be found in the settings section of the Azure portal.
+- **STORAGE_CONNECTION_STRING** must be set to an Azure Storage connection string if you want to test bulk import/export of device identities
+- **IOTHUB_X509_DEVICE_ID** must be set to the name of a device that uses x509 certs for authentication.
+- **IOTHUB_X509_CERTIFICATE** must be set to the path of the x509 certificate pem file for this device.
+- **IOTHUB_X509_KEY** must be set to the path of the x509 certificate key pem file for this device.
+- **IOTHUB_X509_PASSPHRASE** must be set to the passphrase of the x509 certificate key for this device.
+- **IOTHUB_CA_ROOT_CERT** must be set to the *content* of the certificate set a a custom CA cert provisioned on IoT Hub.
```

Comment:
>IOTHUB_CA_ROOT_CERT [](start = 4, length = 19)

Actually this points to a base 64 encoded version of the pem cert file.  We do this so that we don't have to worry about placing \r \n into environment variables.